src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,159 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        private Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            addHead(value);
+        } else {
+            addTail(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> currentNode = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+            if (currentNode.prev != null) {
+                currentNode.prev.next = newNode;
+            } else {
+                head = newNode;
+            }
+            currentNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNodeByIndex(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T value = getNodeByIndex(index).value;
+        getNodeValueByIndexToDelete(index);
+        return value;
     }
 
     @Override
     public boolean remove(T object) {
+        int index = 0;
+        for (Node<T> removedNade = head; removedNade != null; removedNade = removedNade.next) {
+            if (object == null ? removedNade.value == null : object.equals(removedNade.value)) {
+                getNodeValueByIndexToDelete(index);
```

Comment:
Maybe a standard, familiar solution is better? `first == second || first != null && first.equals(second);` and put it into a separate method